engine/games/voltorb_flip.asm

```diff
@@ -0,0 +1,1617 @@
+DEF VOLTORB_FLIP_INTENSE_DELAY_FRAMES EQU 120 ; 2 seconds
+DEF VOLTORB_FLIP_LEVEL_CLEAR_DELAY_FRAMES EQU 60 ; 1 second
+DEF VOLTORB_FLIP_GAME_OVER_DELAY_FRAMES EQU 240 ; 4 seconds
+DEF VOLTORB_FLIP_TILE_UL_BLANK EQU $00
+DEF VOLTORB_FLIP_TILE_UM_BLANK EQU $01
+DEF VOLTORB_FLIP_TILE_UR_BLANK EQU $02
+DEF VOLTORB_FLIP_TILE_ML_BLANK EQU $10
+DEF VOLTORB_FLIP_TILE_MM_BLANK EQU $11
+DEF VOLTORB_FLIP_TILE_MR_BLANK EQU $12
+DEF VOLTORB_FLIP_TILE_LL_BLANK EQU $20
+DEF VOLTORB_FLIP_TILE_LM_BLANK EQU $21
+DEF VOLTORB_FLIP_TILE_LR_BLANK EQU $22
+DEF VOLTORB_FLIP_TILE_UL_MARK EQU $06
+DEF VOLTORB_FLIP_TILE_UR_MARK EQU $08
+DEF VOLTORB_FLIP_TILE_LL_MARK EQU $16
+DEF VOLTORB_FLIP_TILE_LR_MARK EQU $18
+DEF VOLTORB_FLIP_TILE_VOLTORB_TOP_BASE    EQU $03
+DEF VOLTORB_FLIP_TILE_VOLTORB_MIDDLE_BASE EQU $13
+DEF VOLTORB_FLIP_TILE_VOLTORB_BOTTOM_BASE EQU $23
+DEF VOLTORB_FLIP_TILE_CARD_NUMBER_BASE EQU $26
+DEF VOLTORB_FLIP_TILE_FLIP_TOP_BASE EQU $30
+DEF VOLTORB_FLIP_TILE_FLIP_MID_BASE EQU $40
+DEF VOLTORB_FLIP_TILE_FLIP_BOTTOM_BASE EQU $50
+DEF VOLTORB_FLIP_COUNTER_ATTR_OFFSET EQU VOLTORB_FLIP_NUMBER1_TILE_BASE - VOLTORB_FLIP_NUMBER2_TILE_BASE
+DEF VOLTORB_FLIP_COUNTER_PRINT_BC EQU $0102
+DEF VOLTORB_FLIP_COIN_PRINT_BC EQU $0204
+
+; wVoltorbFlipCardProperties
+	const_def
+	const_skip                                ; 0
+	const_skip                                ; 1
+	shift_const VOLTORB_FLIP_CARD_FLIPPED     ; 2
+	shift_const VOLTORB_FLIP_CARD_HIGHLIGHTED ; 3
+	shift_const VOLTORB_FLIP_CARD_MARKED3     ; 4
+	shift_const VOLTORB_FLIP_CARD_MARKED2     ; 5
+	shift_const VOLTORB_FLIP_CARD_MARKED1     ; 6
+	shift_const VOLTORB_FLIP_CARD_MARKED0     ; 7
+
+DEF VOLTORB_FLIP_CARD_NUMBER_MASK EQU %00000011
+
+; Card Properties (Bits):
+; 7-marked 0
+; 6-marked 1
+; 5-marked 2
+; 4-marked 3
+; 3-highlighted
+; 2-Card Flipped
+; 1,0-Card number
+
+; States:
+; sounds:
+; SFX_READ_TEXT move cursor
+; SFX_QUIT_SLOTS game over
+; SFX_INTRO_SUICUNE_1 flip card
+; SFX_CHOOSE_A_CARD are you sure?
+; SFX_DEX_FANFARE_LESS_THAN_20 quit
+; SFX_3RD_PLACE level clear
+; SFX_TRANSACTION coin
+; SFX_EGG_BOMB boom!
+
+VoltorbFlipGFX:
+INCBIN "gfx/voltorb_flip/voltorb_flip.2bpp"
+
+VoltorbFlipPalette:
+	RGB 31,25,01, 31,31,31, 04,22,09, 00,00,00
+	RGB 31,25,01, 31,15,04, 04,22,09, 00,00,00
+	RGB 31,00,00, 31,31,31, 04,22,09, 00,00,00
+	RGB 31,00,00, 31,31,31, 31,25,01, 00,00,00
+	RGB 31,00,00, 31,31,31, 00,20,29, 00,00,00
+	RGB 31,00,00, 31,31,31, 20,09,20, 00,00,00
+	RGB 31,00,00, 31,31,31, 09,27,14, 00,00,00
+	RGB 31,00,00, 31,15,04, 09,27,14, 00,00,00
+
+VoltorbFlipPaletteBlank:
+	RGB 31,31,31, 31,31,31, 31,31,31, 31,31,31
+	RGB 31,31,31, 31,31,31, 31,31,31, 31,31,31
+	RGB 31,31,31, 31,31,31, 31,31,31, 31,31,31
+	RGB 31,31,31, 31,31,31, 31,31,31, 31,31,31
+	RGB 31,31,31, 31,31,31, 31,31,31, 31,31,31
+	RGB 31,31,31, 31,31,31, 31,31,31, 31,31,31
+	RGB 31,31,31, 31,31,31, 31,31,31, 31,31,31
+	RGB 31,31,31, 31,31,31, 31,31,31, 31,31,31
+
+_VoltorbFlip::; this is where the magic happens
```

Comment:
Just checking, has optimize.py been run on this, or is it a direct port of FroggestSpirit's code?